src/main/java/de/cyklon/reflection/entities/impl/ReflectClassImpl.java

```diff
@@ -273,23 +273,35 @@ public String getTypeName() {
 	public String getName() {
 		if (isWildcard()) return "?";
 
-		if (!clazz.isArray()) return clazz.getSimpleName();
+		if (!clazz.isArray()) return clazz.getName();
 		else return getArrayInfo().component().getName() + "[]";
 	}
 
 	@NotNull
 	@Override
-	public String getFullName() {
-		return getTypeName();
+	public String getSimpleName() {
+		return clazz.getSimpleName();
 	}
 
+
 	@Override
 	public boolean equals(Object obj) {
 		return obj instanceof ReflectClass<?> rc && rc.getType().equals(type);
 	}
 
 	@Override
 	public String toString() {
-		return getFullName();
+		return getTypeName();
+	}
+
+
+	@Override
```

Comment:
These methods should probably be implemented in the ReflectClass interface via default methods (isLoaded and load)